app/scripts/lib/createEvmMethodsToNonEvmAccountReqFilterMiddleware.test.ts

```diff
@@ -0,0 +1,235 @@
+import { jsonrpc2 } from '@metamask/utils';
+import { BtcAccountType, EthAccountType } from '@metamask/keyring-api';
+import { Json } from 'json-rpc-engine';
+import createEvmMethodsToNonEvmAccountReqFilterMiddleware, {
+  EvmMethodsToNonEvmAccountFilterMessenger,
+} from './createEvmMethodsToNonEvmAccountReqFilterMiddleware';
+
+describe('createEvmMethodsToNonEvmAccountReqFilterMiddleware', () => {
+  const getMockRequest = (method: string, params?: Json) => ({
+    jsonrpc: jsonrpc2,
+    id: 1,
+    method,
+    params,
+  });
+  const getMockResponse = () => ({ jsonrpc: jsonrpc2, id: 'foo' });
+
+  // @ts-expect-error This function is missing from the Mocha type definitions
+  it.each([
```

Comment:
Ok, this whole list is being repeated twice with only some `boolean` being different, WDYT about doing something like:

```ts
const METHOD_FILTERS = {
    {
      method: 'eth_accounts',
      evm: {
        calledNext: false,
      },
      nonEvm: {
        calledNext: true,
      },
    },
    ...
]
```

So that, we can still use those in a `it.each` and just destructure each objects differently to use either `evm` or `nonEvm` depending on the test case?